pkg/epp/flowcontrol/registry/registry.go

```diff
@@ -683,64 +539,62 @@ func (fr *FlowRegistry) repartitionShardConfigsLocked() {
 
 // --- Internal Helpers ---
 
-// flowComponents holds the set of plugin instances created for a single shard.
+// flowComponents holds the plugin instances created for a single flow on a single shard.
 type flowComponents struct {
 	policy framework.IntraFlowDispatchPolicy
 	queue  framework.SafeQueue
 }
 
 // buildFlowComponents instantiates the necessary plugin components for a new flow instance.
 // It creates a distinct instance of each component for each shard to ensure state isolation.
-func (fr *FlowRegistry) buildFlowComponents(spec types.FlowSpecification, numInstances int) ([]flowComponents, error) {
-	priority := spec.Key.Priority
-	bandConfig, err := fr.config.getBandConfig(priority)
+func (fr *FlowRegistry) buildFlowComponents(key types.FlowKey, numInstances int) ([]flowComponents, error) {
+	bandConfig, err := fr.config.getBandConfig(key.Priority)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get configuration for priority %d: %w", priority, err)
```

Comment:
The error returned from `getBandConfig` is already well-formed. I was doing redundant error wrapping.